scripts/build-mlir.sh

```diff
@@ -52,7 +52,7 @@ while [[ "$#" -ge 1 ]] ; do
 			commit >&1
 			exit 0
 			;;
-		--help)
+		--help|*)
```

Comment:
Out of curiosity: What does this do?

Reply:
Is this the "avoid looping indefinitely"?

Reply:
@phate yes, this switch statement is inside a while loop that keeps running until all arguments have been shifted. If nothing matches, nothing is shifted, and it loops forever